src/analytics.js

```diff
@@ -59,38 +60,82 @@ export function triggerAnalyticsEvent(nodeOrDoc, eventType, opt_vars) {
   });
 }
 
-/**
- * Method to create scoped analytics element for any element.
- * @param {!Element} parentElement
- * @param {!JSONType} config
- * @param {boolean=} loadAnalytics
- */
-export function insertAnalyticsElement(
-    parentElement, config, loadAnalytics = false) {
-  const doc = parentElement.ownerDocument;
-  const analyticsElem = doc.createElement('amp-analytics');
-  analyticsElem.setAttribute('sandbox', 'true');
-  const scriptElem = createElementWithAttributes(doc,
-        'script', {
-          'type': 'application/json',
-        });
-  scriptElem.textContent = JSON.stringify(config);
-  analyticsElem.appendChild(scriptElem);
-  analyticsElem.CONFIG = config;
+export class ExtensionAnalytics {
```

Comment:
Yes, I get it is very like creating a simplified version of analytics root. However I can't skip this workaround because I will need info to map the custom trigger event to the correct handler when 3p extension triggers it. This class let them only `#triggerEventForTarget` with target set to be the created `<amp-analytics>` element, thus provide me a way to map `<amp-analytics>` element id to the corresponding handler. TBH I am super convinced  creating analytics root should be the solution here..... Eh, before V2, what's our plan here then...